src/Json.Schema.ToDotNet.UnitTests/DataModelGeneratorTests.cs

```diff
@@ -2363,7 +2369,152 @@ public int GetHashCode(C obj)
                 ["D"] = new ExpectedContents()
             };
 
-            Assert.FileContentsMatchExpectedContents(_testFileSystem, expectedContentsDictionary);
+            VerifyGeneratedFileContents(expectedContentsDictionary);
+        }
+
+        [Fact(DisplayName = "DataModelGenerator generates attributes for properties with defaults.")]
+        public void GeneratesAttributesForPropertiesWithDefaults()
+        {
+            const string Schema =
+@"{
+  ""type"": ""object"",
+  ""properties"": {
+    ""intProp"": {
+      ""type"": ""integer"",
+      ""description"": ""An integer property.""
+    },
+    ""intPropWithDefault"": {
+      ""type"": ""integer"",
+      ""description"": ""An integer property with a default value."",
+      ""default"": 42
+    },
+    ""numberProp"": {
+      ""type"": ""number"",
+      ""description"": ""A number property.""
+    },
+    ""numberPropWithDefault"": {
+      ""type"": ""number"",
+      ""description"": ""A number property with a default value."",
+      ""default"": 42.1
+    },
+    ""stringProp"": {
+      ""type"": ""string"",
+      ""description"": ""A string property.""
+    },
+    ""stringPropWithDefault"": {
+      ""type"": ""string"",
+      ""description"": ""A string property with a default value."",
+      ""default"": ""42""
+    },
+    ""boolProp"": {
+      ""type"": ""boolean"",
+      ""description"": ""A Boolean property.""
+    },
+    ""boolPropWithTrueDefault"": {
+      ""type"": ""boolean"",
+      ""description"": ""A Boolean property with a true default value."",
+      ""default"": true
+    },
+    ""boolPropWithFalseDefault"": {
+      ""type"": ""boolean"",
+      ""description"": ""A Boolean property with a false default value."",
+      ""default"": false
```

Comment:
>false [](start = 19, length = 5)

not sure how interesting this is, since it's the .NET default. why did you include it? if this test is interesting, why not add a string property with the default of 'null'? and an integer with 0, etc? just to be comprehensive.

Reply:
It's only there because it exercises a code branch. In Roslyn the `true` literal is a different `SyntaxKind` from the `false` literal, so there's a ternary operator in the code to generate one or the other. There's no similar branch for numeric values. And you can't put `null` for a `string`-valued property; `null` is a different data type.

---
In reply to: [244508342](https://github.com/Microsoft/jschema/pull/62#discussion_r244508342) [](ancestors = 244508342)